### ISSUE TITLE
what lengths are supported for frameshift indels

### ISSUE THREAD

**Issue by ieiwk** [jamesware/denovolyzeR#14]
are only single nucleotide indels supported? Thanks!

**Comment by jamesware** [comment]
The expected rate of frameshift indels is not modelled in the same way as the SNPs, which are based on nucleotide spontaneous mutation rates.
Indels were modelled (as described in Samocha 2014) as a ratio to the rate of nonsense mutations - an empirical estimate that fitted observed control data reasonably well.

Frameshift indels for this estimate were not restricted in length - it was essentially whatever was identified by the standard small variant callers (GATK in practice), and annotated as fs by the ensembl VEP.

So - any "small indel" is supported.  It is always worth checking that the model is a good fit to data from control trios sequenced and analysed using the platforms and pipeline that you are using in your case series.

**Comment by ieiwk** [comment]
OK. I thought only one-nucleotide indels count. Thanks a lot!

**Comment by ieiwk** [comment]
But wait. In the paper "Interpreting de novo Variation in Human Disease Using denovolyzeR", in "COMMENTARY" section, about "Large variants", it was wrote that "This analytical framework assesses single nucleotide variants and small (single basepair) insertions and deletions only. Larger insertions, deletions, and other structural variants are not assessed.".

So should I exclude frameshift indels that involve 2 or more nucleotides?

**Comment by jamesware** [comment]
I checked with Kaitlin Samocha directly.  She replied that she was using all frameshift indels from a small indel calling pipeline (GATK), and not restricting strictly to single bp.  It does not include larger SVs that require other detection techniques.
However, since this part of the model is an approximation, I suggest that the key is to assess whether it is a good fit to your control data.

**Comment by ieiwk** [comment]
I see. We have no control at all... That's why we rely heavily on this model. I checked the reported de novo SNVs from 4105 normal controls from many exome or genome sequencing studies collected at the denovo-db website (http://denovo-db.gs.washington.edu/denovo-db/), but I found only ~0.5 de novo SNV per exome. This number is too small compared to current consensus. In one PNAS paper of this year I saw this number being 0.7+ (the authors used data from several large neural develomental studies), which is also substantially smaller than 1.

So now I think this tool is very useful. And thanks a lot for your being responsive :)
